src/z3c/menu/ready2go/checker.py

```diff
@@ -68,36 +68,33 @@ def selected(self):
 # default selected checkers
 class GlobalSelectedChecker(FalseSelectedChecker):
     """Global menu item selected checker.
-    
+
     Note, this is a menu group which is selected on different menu items.
     You need to register for each view a TrueSelectedChecker if the site menu
     item should get rendered as selected.
     """
 
     zope.component.adapts(zope.interface.Interface, IBrowserRequest,
-        zope.interface.Interface, interfaces.IMenuManager,
-        interfaces.IGlobalMenuItem)
+                          zope.interface.Interface, interfaces.IMenuManager,
+                          interfaces.IGlobalMenuItem)
```

Comment:
Wow, does this still work on Python 3?  I thought we were supposed to use @zope.component.adapter(...) as a class decorator nowadays.

(Out of scope for this PR)

Reply:
It does, yes, the same as `taggedValue()` still works in an interface body. Both of those work by just placing extra items into the class namespace using `sys._getframe().f_locals`.

What made using `zope.interface.implements()` not work in a class body is that [it tries to assign to `__metaclass__` in the class body](https://github.com/zopefoundation/zope.interface/blob/bf0de08af2d53261d7eb0ce1e31f574f93524ef6/src/zope/interface/advice.py#L82) using the same trick. Since `__metaclass__` isn't used at all in Python 3, that's a non-starter.

TBH, I'm not entirely sure why it does that. (At a guess, it may be trying to collapse the list of implemented interfaces including from super classes at class creation time?)